docs/technical/git-submodules.md

```diff
@@ -0,0 +1,167 @@
+- Changing how Git submodules are implemented in Git
+
+## Background
+
+We mainly want to support Git submodules for feature parity, since Git
+submodules are a standard feature in Git and are popular enough that we have
+received user requests for them. Secondarily (and distantly so), Git submodules
+are notoriously difficult to use, so there is an opportunity to improve the UX
+over Git's implementation.
+
+### Git Submodules 101
+
+[Git submodules](https://git-scm.com/docs/gitsubmodules) are a feature of Git
+that allow a repository (submodule) to be embedded inside another repository
+(the superproject). Notably, a submodule is a full repository, complete with its
+own index, object store and ref store. It can be interacted with like any other
+repository, regardless of the superproject.
+
+In a superproject commit, submodule information is captured in two places:
+
+- A `gitlink` entry in the commit's tree, where the value of the `gitlink` entry
+  is the submodule commit id. This tells Git what to populate in the working
+  tree.
+
+- A top level `.gitmodules` file. This file is in Git's config syntax and
+  entries take the form `submodule.<submodule-name>.*`. These include many
+  settings about the submodules, but most importantly:
+
+  - `submodule<submodule-name>.path` contains the path from the root of the tree
+    to the `gitlink` being described.
+
+  - `submodule<submodule-name>.url` contains the url to clone the submodule
+    from.
+
+In the working tree, Git notices the presence of a submodule by the `.git` entry
+(signifying the root of a Git repository working tree). This is either the
+submodule's actual Git directory (an "old-form" submodule), or a `.git` file
+pointing to `<superproject-git-directory>/modules/<submodule-name>`. The latter
+is sometimes called the "absorbed form", and is Git's preferred mode of
+operation.
+
+## Roadmap
+
+Git submodules should be implemented in an order that supports an increasing set
+of workflows, with the goal of getting feedback early and often. When support is
+incomplete, jj should not crash, but instead provide fallback behavior and warn
+the user where needed.
+
+The goal is to land good support for pure Jujutsu repositories, while colocated
+repositories will be supported when convenient.
+
+This section should be treated as a set of guidelines, not a strict order of
+work.
+
+### Phase 1: Readonly submodules
+
+This includes work that inspects submodule contents but does not create new
+objects in the submodule. This requires a way to store submodules in a jj
+repository that supports readonly operations.
+
+#### Outcomes
+
+- Submodules can be cloned anew
+- New submodule commits can be fetched
+- Submodule history and branches can be viewed
+- Submodule contents are populated in the working copy
+- Superproject gitlink can be updated to an existing submodule commit
+- Conflicts in the superproject gitlink can be resolved to an existing submodule
+  commit
```

Comment:
Good question. I don't really like the inconsistency either; I use it out of habit, and the Git docs say "superproject" if they say anything at all. My preference would be to stick with Git terminology for the Git-specific thing so that we'll leave the nicer terminology alone in case we ever implement generic jj subrepos.